docs/proposals/20210323-etcdadm-bootstrap-provider.md

```diff
@@ -0,0 +1,556 @@
+---
+title: Add support for managed external etcd clusters in CAPI
+authors:
+  - "@mrajashree"
+creation-date: 2021-03-23
+last-updated: 2021-03-23
+status: provisional
+---
+
+# Add support for managed external etcd clusters in CAPI
+
+## Table of Contents
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [POC implementation](#poc-implementation)
+    - [Questions/Concerns](#questions/concerns-about-implementation)
+    - [Design questions](#design-questions)
+    - [Security Model](#security-model)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+    - [Graduation Criteria [optional]](#graduation-criteria-optional)
+    - [Version Skew Strategy [optional]](#version-skew-strategy-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+If this proposal adds new terms, or defines some, make the changes to the book's glossary when in PR stage.
+
+## Summary
+
+This is a proposal for adding support for provisioning external etcd clusters in Cluster API. CAPI's KubeadmControlPlane supports using an external etcd cluster. However, it currently does not support provisioning and managing this external etcd cluster. 
+As per [this KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/etcdadm/2496-etcdadm), there are ongoing
+efforts to rebase [etcd-manager](https://github.com/kopeio/etcd-manager) (used in kOps) on to etcdadm, so as to make etcdadm a consistent etcd solution to be used across all projects, including cluster-api.
```

Comment:
So if I understand this correctly, `etcd-manager` acts as a control plane which does node placement and orchestrates all config updates (upgrade), whereas `etcdadm` is an agent launched on each node?

Reply:
Clearly defining the scope of each component will help understand the whole workflow at high level.